pkg/resources/kafka/kafka.go

```diff
@@ -359,6 +258,128 @@ func (r *Reconciler) Reconcile(log logr.Logger) error {
 	return nil
 }
 
+func (r *Reconciler) reconcileKafkaPodDelete(log logr.Logger) error {
```

Comment:
I would rename this "handleBrokersDownscale" as it's checks the  brokers list in the spec.
I am thinking we can add in a future PR a check here to limit the dowscale under a threshold also

Reply:
Good idea, we are going to address this on a different commit, since all our reconcile method name follows this pattern, we are going to rename all to handleBrokersUpscale,Downscale